tools/extra/fpgadiag/fpgastats.py

```diff
@@ -0,0 +1,189 @@
+                    ('CNTR_RX_PAUSE_ERR', 0x914, 2),
+                    ('CNTR_RX_64B', 0x916, 2),
+                    ('CNTR_RX_65to127B', 0x918, 2),
+                    ('CNTR_RX_128to255B', 0x91a, 2),
+                    ('CNTR_RX_256to511B', 0x91c, 2),
+                    ('CNTR_RX_512to1023B', 0x91e, 2),
+                    ('CNTR_RX_1024to1518B', 0x920, 2),
+                    ('CNTR_RX_1519toMAXB', 0x922, 2),
+                    ('CNTR_RX_OVERSIZE', 0x924, 2),
+                    ('CNTR_RX_MCAST_DATA_OK', 0x926, 2),
+                    ('CNTR_RX_BCAST_DATA_OK', 0x928, 2),
+                    ('CNTR_RX_UCAST_DATA_OK', 0x92a, 2),
+                    ('CNTR_RX_MCAST_CRTL', 0x92c, 2),
+                    ('CNTR_RX_BCAST_CTRL', 0x92e, 2),
+                    ('CNTR_RX_UCAST_CTRL', 0x930, 2),
+                    ('CNTR_RX_PAUSE', 0x932, 2),
+                    ('CNTR_RX_RUNT', 0x934, 2))
+    fifo_stats_10g = (('MUX_CDC_FIFO_CNTR_FULL', 0x1, 1),
+                      ('MUX_CDC_FIFO_CNTR_ERROR', 0x2, 1),
+                      ('MUX_CDC_FIFO_CNTR_SOP_MISSED', 0x3, 1),
+                      ('MUX_CDC_FIFO_CNTR_EOP_MISSED', 0x4, 1),
+                      ('DEMUX_CDC_FIFO_CNTR_FULL', 0x101, 1),
+                      ('DEMUX_CDC_FIFO_CNTR_ERROR', 0x102, 1),
+                      ('DEMUX_CDC_FIFO_CNTR_SOP_MISSED', 0x103, 1),
+                      ('DEMUX_CDC_FIFO_CNTR_EOP_MISSED', 0x104, 1))
+    fifo_stats_25_40g = (('MUX_CDC_FIFO_CNTR_FULL', 0x1, 1),
+                         ('MUX_CDC_FIFO_CNTR_ERROR', 0x2, 1),
+                         ('MUX_CDC_FIFO_CNTR_SOP_MISSED', 0x3, 1),
+                         ('MUX_CDC_FIFO_CNTR_EOP_MISSED', 0x4, 1),
+                         ('DEMUX_CDC_FIFO_CNTR_FULL', 0x41, 1),
+                         ('DEMUX_CDC_FIFO_CNTR_ERROR', 0x42, 1),
+                         ('DEMUX_CDC_FIFO_CNTR_SOP_MISSED', 0x43, 1),
+                         ('DEMUX_CDC_FIFO_CNTR_EOP_MISSED', 0x44, 1))
+
+    def __init__(self, args):
+        self.eth_grps = args.eth_grps
+        self.mac_number = 0
+
+    def print_device_stats(self, handler, stats, reg, length, fifo=False):
+        print("{0: <30}".format(stats), end=' | ')
+        comp = 'eth' if fifo else 'mac'
+        for i in range(self.mac_number):
+            data = 0
+            for l in range(length):
+                addr = reg + l * 8 if fifo else (reg + l)
+                dev = 0 if fifo else i
+                v = self.fpga_eth_reg_read(handler, comp, dev, addr)
+                data += (0xffffffff & v) << (32 * l)
+            print("{0: <12}".format(data), end=' | ')
+
+    def print_stats(self, info):
+        for w in info:
+            _, self.mac_number, spd, node = info[w]
+            print('=' * 100)
+            print('MAC wrapper {}, Speed {}g'.format(w, spd))
+            print("{0: <30}".format('stats'), end=' | ')
+            for i in range(self.mac_number):
+                print('mac {:<8}'.format(i), end=' | ')
```

Comment:
I ran this file and the output is very difficult to follow/comprehend. Can the output be refactored for better readability.